### DISCUSSION TITLE
How to pass appsecret_proof parameter

### DISCUSSION THREAD

**Discussion by rv5gestao** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#42] [General]
Hello Gabriel,

In my code:
var messageType = Convert.ToString(messageReceived["entry"][0]["changes"][0]["value"]["messages"][0]["type"]);
                    var name = Convert.ToString(messageReceived["entry"][0]["changes"][0]["value"]["contacts"][0]["profile"]["name"]);
                    string phoneNumberWhatsapp = Convert.ToString(messageReceived["entry"][0]["changes"][0]["value"]["metadata"]["display_phone_number"]);
                    string phoneNumberId = Convert.ToString(messageReceived["entry"][0]["changes"][0]["value"]["metadata"]["phone_number_id"]);

                    var estabModel = EstablishmentBO.Get EstablishmentByWhatsappNumber(phoneNumberWhatsapp);
                    var atendimentoPeloiBot = estabModel.EstabelecimentoConfiguracao.IdFormaAtendimento == 0;

                    string IBot response;
                    if (messageType.Equals("text"))
                    {
                        var textMessageReceived = JsonConvert.DeserializeObject<TextMessageReceived>(Convert.ToString(messageReceived)) as TextMessageReceived;
                        textMessage = new List<TextMessage>(textMessageReceived.Entry.SelectMany(x => x.Changes).SelectMany(x => x.Value.Messages));

                        MarkMessageRequest markMessageRequest = new()
                        {
                            MessageId = textMessage.SingleOrDefault().Id,
                            Status = "read"
                        };

                        await whatsAppBusinessClient.MarkMessageAsReadAsync(markMessageRequest);

When "await whatsAppBusinessClient.MarkMessageAsReadAsync(markMessageRequest);" is called, it returns this error message:

Error Message: API calls from the server require an appsecret_proof argument

I haven't found a way to pass this parameter.

How can I resolve this?

**Comment by rv5gestao** [comment]
Help me!

**Reply by gabrieldwight** [reply]
Hi, does your facebook app have app secret enabled from advance settings? I have not yet tested with app secret settings.

**Reply by rv5gestao** [reply]
Yes, I disabled it and it started working.
Thanks!
